packages/editor-ui/src/components/WorkflowActivator.test.ts

```diff
@@ -0,0 +1,80 @@
+import { describe, it, expect, vi } from 'vitest';
+import WorkflowActivator from '@/components/WorkflowActivator.vue';
+import userEvent from '@testing-library/user-event';
+
+import { useWorkflowsStore } from '@/stores/workflows.store';
+
+import { createTestingPinia } from '@pinia/testing';
+import { createComponentRenderer } from '@/__tests__/render';
+import { mockedStore } from '@/__tests__/utils';
+import { EXECUTE_WORKFLOW_TRIGGER_NODE_TYPE } from '@/constants';
+
+const renderComponent = createComponentRenderer(WorkflowActivator);
+let mockWorkflowsStore: ReturnType<typeof mockedStore<typeof useWorkflowsStore>>;
+
+describe('WorkflowActivator', () => {
+	beforeEach(() => {
+		createTestingPinia();
+
+		mockWorkflowsStore = mockedStore(useWorkflowsStore);
+	});
+
+	afterEach(() => {
+		vi.clearAllMocks();
+	});
+
+	it('it renders correctly', () => {
+		const renderOptions = {
+			props: {
+				workflowActive: false,
+				workflowId: '1',
+				workflowPermissions: { update: true },
+			},
+		};
+
+		expect(() => renderComponent(renderOptions)).not.toThrow();
```

Comment:
Agree with @cstuncsik but we can be more explicit in this test. Since the goal of this component is to render a label and a toggle, we can test if it renders correctly like this:
```ts
const { getByTestId, getByRole } = renderComponent(renderOptions);
expect(getByTestId('workflow-activator-status')).toBeInTheDocument();
expect(getByRole('switch')).toBeInTheDocument();
```

We can leave the `.not.toThrow()` test alongside this so we make sure we catch rendering errors